services/galley/src/Galley/API/Teams.hs

```diff
@@ -453,9 +454,14 @@ uncheckedAddTeamMember tid nmem = do
   -- FUTUREWORK: We cannot enable legalhold on large teams right now
   ensureNotTooLargeForLegalHold tid mems
   (TeamSize sizeBeforeAdd) <- addTeamMemberInternal tid Nothing Nothing nmem mems
-  -- FUTUREWORK: This takes a list of members for fanout, but it should really
-  --             use the hardlimit instead. We want to avoid double lookups though
-  Journal.teamUpdate tid (sizeBeforeAdd + 1) $ newTeamMemberList ((nmem ^. ntmNewTeamMember) : mems ^. teamMembers) (mems ^. teamMemberListType)
+  let new = nmem ^. ntmNewTeamMember
+
+  -- This has to only be done for binding teams, it is assumed here as this function is
+  -- supposed to be called only for binding teams
+  when (hasPermission new SetBilling) $
+    Data.addBillingTeamMember tid (new ^. userId)
```

Comment:
Same comment as above; when adding/removing/updating team members, we should update this table inside the data module